tests/integration/test_http_get_all.py

```diff
@@ -0,0 +1,88 @@
+    * Verify requests can be found using by using the state=complete filter.
+    Checks:
+    * Check that response code is 200
+    * Check that filtering parameters are working properly (page, per_page, and state)
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    requests_amount = test_env["http_get_all"]["requests_amount"]
+    submitted_requests = []
+    for _ in range(requests_amount):
+        initial_response = client.create_new_request(
+            payload={
+                "repo": test_env["package"]["repo"],
+                "ref": test_env["package"]["ref"],
+                "pkg_managers": test_env["package"]["pkg_managers"],
+            },
+        )
+        completed_response = client.wait_for_complete_request(initial_response)
+        response_data = completed_response.data
+        submitted_requests.append(response_data["id"])
+
+    assert_no_requests_in_progress_state(client, requests_amount, submitted_requests)
+    assert_completed_requests(client, requests_amount)
+
+
+def test_get_all_verbose(test_env, default_request, tmpdir):
+    """
+    Check the API endpoint to get all requests with the verbose flag enabled.
+
+    Process:
+    * Create a request and wait for it to complete
+    * Send request get_all to the Cachito API with verbose
+    Checks:
+    * Check that response code is 200
+    * Check that verbose is working properly
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    initial_response = client.create_new_request(
+        payload={
+            "repo": test_env["package"]["repo"],
+            "ref": test_env["package"]["ref"],
+            "pkg_managers": test_env["package"]["pkg_managers"],
+        },
+    )
+    client.wait_for_complete_request(initial_response)
+
+    request_json = {"per_page": 1, "page": 1, "state": "complete", "verbose": True}
+    response = client.fetch_all_requests(request_json)
+    assert response.status == 200
+
+    request_data = response.data["items"][0]
+    request_id = request_data["id"]
+    expected_request_data = client.fetch_request(request_id).data
+    assert request_data == expected_request_data
+
+
+def assert_no_requests_in_progress_state(client: utils.Client, requests_amount, submitted_requests):
+    """Check get_all request returns expected in_progress values."""
+    request_json = {"per_page": requests_amount, "page": 1, "state": "in_progress"}
+    response = client.fetch_all_requests(request_json)
+    assert response.status == 200
+
+    assert all(item["state"] == "in_progress" for item in response.data["items"])
+
+    response_ids = set(item["id"] for item in response.data["items"])
+    assert not any(request_id in response_ids for request_id in submitted_requests)
+    assert 0 <= len(response.data["items"]) <= requests_amount
+
+
+def assert_completed_requests(client: utils.Client, requests_amount):
```

Comment:
Could you also please verify that the submitted requests appear in in `items`?

Reply:
I am not sure we need it. We are sending requests because the cachito log can be empty. In case of network delay we can get the problem: 
1. we submitted 2 requests
2. somebody submitted 2 requests
3. we are getting  2 requests and expecting requests from 1. But we will get requests from 2.

Reply:
We are just checking that HTTP GET ALL with 'state=complete' parameter returns completed requests.

Reply:
Done!